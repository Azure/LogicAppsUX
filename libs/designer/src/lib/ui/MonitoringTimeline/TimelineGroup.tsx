import { Button, Text } from '@fluentui/react-components';
import { useEffect, useMemo, useState } from 'react';
import { useMonitoringTimelineStyles } from './monitoringTimeline.styles';
import type { TimelineRepetitionWithActions } from './helpers';
import { bundleIcon, ChevronDown24Filled, ChevronDown24Regular, ChevronRight24Filled, ChevronRight24Regular } from '@fluentui/react-icons';
import { useIntl } from 'react-intl';
import TimelineNode from './TimelineNode';

const ExpandIcon = bundleIcon(ChevronRight24Filled, ChevronRight24Regular);
const CollapseIcon = bundleIcon(ChevronDown24Filled, ChevronDown24Regular);

interface TimelineGroupProps {
  isTimelineExpanded: boolean;
  taskId: number;
  repetitions: TimelineRepetitionWithActions[];
  transitionIndex: number;
  selectedRepetition: TimelineRepetitionWithActions | null;
  handleSelectRepetition: (groupIndex: number, repetitionIndex: number) => void;
}

const TimelineGroup = ({
  isTimelineExpanded,
  taskId,
  repetitions,
  transitionIndex,
  selectedRepetition,
  handleSelectRepetition,
}: TimelineGroupProps) => {
  const [isGroupExpanded, setIsGroupExpanded] = useState(false);
  const styles = useMonitoringTimelineStyles();
  const intl = useIntl();

  const handleToggleExpand = (): void => {
    setIsGroupExpanded(!isGroupExpanded);
  };

  const text = useMemo(
    () => ({
      taskCount: (count: number) =>
        intl.formatMessage(
          {
            defaultMessage: 'Task #{count}',
            id: 'wbjWVF',
            description: 'Text displaying the task number.',
          },
          { count }
        ),
    }),
    [intl]
  );

  useEffect(() => {
    if (taskId === transitionIndex) {
      setIsGroupExpanded(true);
    }
  }, [transitionIndex, taskId]);

  return (
    <div style={{ display: 'flex', flexDirection: 'column' }}>
      {isTimelineExpanded ? (
        <Button
          className={styles.timelineTask}
          appearance="subtle"
          size="small"
          onClick={handleToggleExpand}
          icon={isGroupExpanded ? <CollapseIcon /> : <ExpandIcon />}
          style={{ justifyContent: 'center', flexGrow: 1 }}
        >
          {isTimelineExpanded ? text.taskCount(taskId + 1) : taskId + 1}
        </Button>
      ) : (
        <Text className={styles.timelineTask} align={'center'} size={200} weight={'medium'}>
          {taskId + 1}
        </Text>
      )}

      {isGroupExpanded &&
        repetitions.map((repetition, index) => {
          return (
            <div key={repetition.repetitionIndex} style={{ margin: '3px' }}>
              <TimelineNode
                selected={repetition.repetitionIndex === selectedRepetition?.repetitionIndex}
                onSelect={() => handleSelectRepetition(taskId, index)}
                isExpanded={isTimelineExpanded}
                data={repetition.data!}
              />
            </div>
          );
        })}
    </div>
  );
};

export default TimelineGroup;
