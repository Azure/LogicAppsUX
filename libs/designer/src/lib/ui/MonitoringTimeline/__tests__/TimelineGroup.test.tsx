import type { ComponentProps } from 'react';
import { describe, it, expect, beforeEach, vi } from 'vitest';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import TimelineGroup from '../TimelineGroup';
import type { TimelineRepetitionWithActions } from '../helpers';

// Mock the styles
vi.mock('../monitoringTimeline.styles', () => ({
  useMonitoringTimelineStyles: () => ({
    timelineTask: 'timeline-task',
  }),
}));

// Mock TimelineNode component
vi.mock('../TimelineNode', () => ({
  default: ({ selected, data }: { selected: boolean; data: any }) => (
    <div data-testid="timeline-node" data-selected={selected}>
      TimelineNode - {data?.id}
    </div>
  ),
}));

type TimelineGroupProps = ComponentProps<typeof TimelineGroup>;

describe('TimelineGroup', () => {
  let defaultProps: TimelineGroupProps;
  const mockHandleSelectRepetition = vi.fn();

  const createMockRepetition = (repetitionIndex: number, taskId: number, actionId = 'action1'): TimelineRepetitionWithActions => ({
    actionIds: [actionId],
    repetitionIndex,
    data: {
      id: `rep-${repetitionIndex}`,
      name: `repetition-${repetitionIndex}`,
      properties: {
        actions: {
          [actionId]: {
            status: 'succeeded',
          },
        },
        canResubmit: false,
        correlation: '',
        startTime: '2024-01-01T00:00:00Z',
        status: 'succeeded',
        a2ametadata: {
          taskId,
        },
      },
      type: 'repetition',
    },
  });

  beforeEach(() => {
    vi.clearAllMocks();

    defaultProps = {
      isTimelineExpanded: true,
      taskId: 0,
      repetitions: [createMockRepetition(0, 0), createMockRepetition(1, 0), createMockRepetition(2, 0)],
      transitionIndex: 0,
      selectedRepetition: createMockRepetition(1, 0),
      handleSelectRepetition: mockHandleSelectRepetition,
    };
  });

  const renderWithIntl = (props: TimelineGroupProps) => {
    return renderer.create(
      <IntlProvider locale="en" defaultLocale="en">
        <TimelineGroup {...props} />
      </IntlProvider>
    );
  };

  it('should render with default props when timeline is expanded', () => {
    const tree = renderWithIntl(defaultProps).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render when timeline is collapsed', () => {
    const tree = renderWithIntl({
      ...defaultProps,
      isTimelineExpanded: false,
    }).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render as collapsed group initially', () => {
    const tree = renderWithIntl({
      ...defaultProps,
      transitionIndex: 1, // different from taskId (0)
    }).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render as expanded group when taskId matches transitionIndex', () => {
    const tree = renderWithIntl({
      ...defaultProps,
      transitionIndex: 0, // matches taskId (0)
    }).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with different task number', () => {
    const tree = renderWithIntl({
      ...defaultProps,
      taskId: 5,
    }).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with single repetition', () => {
    const tree = renderWithIntl({
      ...defaultProps,
      repetitions: [createMockRepetition(0, 0)],
    }).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with empty repetitions', () => {
    const tree = renderWithIntl({
      ...defaultProps,
      repetitions: [],
    }).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with no selected repetition', () => {
    const tree = renderWithIntl({
      ...defaultProps,
      selectedRepetition: null,
    }).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call handleSelectRepetition when expand button is clicked', () => {
    const component = renderWithIntl({
      ...defaultProps,
      transitionIndex: 1, // different from taskId to start collapsed
    });

    const buttons = component.root.findAllByType('button');
    expect(buttons).toHaveLength(1);

    // Click expand button
    buttons[0].props.onClick();

    // Should not call handleSelectRepetition for expand action
    expect(mockHandleSelectRepetition).not.toHaveBeenCalled();
  });

  it('should call handleSelectRepetition when timeline node is selected', () => {
    const component = renderWithIntl({
      ...defaultProps,
      transitionIndex: 0, // same as taskId to start expanded
    });

    // First need to expand the group manually
    const buttons = component.root.findAllByType('button');
    if (buttons.length > 0) {
      buttons[0].props.onClick(); // Expand the group
    }

    // Re-render to get the updated state
    component.update(
      <IntlProvider locale="en" defaultLocale="en">
        <TimelineGroup {...defaultProps} transitionIndex={0} />
      </IntlProvider>
    );

    // Find TimelineNode components by their mock implementation
    const timelineNodes = component.root.findAllByProps({ 'data-testid': 'timeline-node' });

    if (timelineNodes.length > 0) {
      // The onClick handler is actually on the wrapping div
      const nodeContainer = timelineNodes[0].parent;
      if (nodeContainer?.props.onClick) {
        nodeContainer.props.onClick();
        expect(mockHandleSelectRepetition).toHaveBeenCalledWith(0, 0); // taskId, index
      }
    }
  });

  it('should show selected repetition correctly', () => {
    const selectedRep = createMockRepetition(1, 0);
    const component = renderWithIntl({
      ...defaultProps,
      selectedRepetition: selectedRep,
      transitionIndex: 0, // same as taskId to start expanded
    });

    // First need to expand the group manually since useEffect doesn't trigger in tests the same way
    const buttons = component.root.findAllByType('button');
    if (buttons.length > 0) {
      buttons[0].props.onClick(); // Expand the group
    }

    // Re-render to apply state changes
    component.update(
      <IntlProvider locale="en" defaultLocale="en">
        <TimelineGroup {...defaultProps} selectedRepetition={selectedRep} transitionIndex={0} />
      </IntlProvider>
    );

    const timelineNodes = component.root.findAllByProps({ 'data-testid': 'timeline-node' });

    if (timelineNodes.length >= 2) {
      // Second node should be selected (repetition index 1)
      expect(timelineNodes[1].props['data-selected']).toBe(true);
      expect(timelineNodes[0].props['data-selected']).toBe(false);
    }
  });

  it('should handle transition index changes', () => {
    // Test that component renders correctly with different transition indices
    const component1 = renderWithIntl({
      ...defaultProps,
      transitionIndex: 1, // different from taskId
    });

    const component2 = renderWithIntl({
      ...defaultProps,
      transitionIndex: 0, // same as taskId
    });

    // Both should render without errors
    expect(component1.toJSON()).toBeTruthy();
    expect(component2.toJSON()).toBeTruthy();

    // The components should have different rendered output based on transitionIndex
    expect(component1.toJSON()).not.toEqual(component2.toJSON());
  });
});
