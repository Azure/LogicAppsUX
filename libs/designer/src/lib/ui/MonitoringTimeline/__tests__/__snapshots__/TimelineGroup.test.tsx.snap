// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TimelineGroup > should render as collapsed group initially 1`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <button
    className="fui-Button r1alrhcs timeline-task ___rq2aul0_9etai30 fhovq9v f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb f1t94bn6 f1s2uweq fr80ssc f1ukrpxl fecsdlb fnwyq0v ft1hn21 fuxngvv fy5bs14 fsv2rcd f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1omzyqd f1dfjoow f1j98vj9 fj8yq94 f4xjyn1 f1et0tmh f9ddjv3 f1wi8ngl fh7ncta f1brlhvm f1sl3k7w fneth5b ft85np5 fy9rknc figsok6 fwrc4pm fazmxh"
    onClick={[Function]}
    style={
      {
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
    type="button"
  >
    <span
      className="fui-Button__icon rywnvv2 ___14u6ce1_ubi7uv0 f1nizpg2 fe5j1ua fjamq6b f64fuq3 fbaiahx"
    >
      <svg
        aria-hidden={true}
        className="fui-Icon-filled ___1vjqft9_1uxjcqz fjseox fez10in fg4l7m0"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.3 4.3a1 1 0 0 0 0 1.4l6.29 6.3-6.3 6.3a1 1 0 1 0 1.42 1.4l7-7a1 1 0 0 0 0-1.4l-7-7a1 1 0 0 0-1.42 0Z"
          fill="currentColor"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="fui-Icon-regular ___12fm75w_1vnroxd f1w7gpdv fez10in fg4l7m0"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.47 4.22c-.3.3-.3.77 0 1.06L15.19 12l-6.72 6.72a.75.75 0 1 0 1.06 1.06l7.25-7.25c.3-.3.3-.77 0-1.06L9.53 4.22a.75.75 0 0 0-1.06 0Z"
          fill="currentColor"
        />
      </svg>
    </span>
    Task #1
  </button>
</div>
`;

exports[`TimelineGroup > should render as expanded group when taskId matches transitionIndex 1`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <button
    className="fui-Button r1alrhcs timeline-task ___rq2aul0_9etai30 fhovq9v f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb f1t94bn6 f1s2uweq fr80ssc f1ukrpxl fecsdlb fnwyq0v ft1hn21 fuxngvv fy5bs14 fsv2rcd f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1omzyqd f1dfjoow f1j98vj9 fj8yq94 f4xjyn1 f1et0tmh f9ddjv3 f1wi8ngl fh7ncta f1brlhvm f1sl3k7w fneth5b ft85np5 fy9rknc figsok6 fwrc4pm fazmxh"
    onClick={[Function]}
    style={
      {
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
    type="button"
  >
    <span
      className="fui-Button__icon rywnvv2 ___14u6ce1_ubi7uv0 f1nizpg2 fe5j1ua fjamq6b f64fuq3 fbaiahx"
    >
      <svg
        aria-hidden={true}
        className="fui-Icon-filled ___1vjqft9_1uxjcqz fjseox fez10in fg4l7m0"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.3 4.3a1 1 0 0 0 0 1.4l6.29 6.3-6.3 6.3a1 1 0 1 0 1.42 1.4l7-7a1 1 0 0 0 0-1.4l-7-7a1 1 0 0 0-1.42 0Z"
          fill="currentColor"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="fui-Icon-regular ___12fm75w_1vnroxd f1w7gpdv fez10in fg4l7m0"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.47 4.22c-.3.3-.3.77 0 1.06L15.19 12l-6.72 6.72a.75.75 0 1 0 1.06 1.06l7.25-7.25c.3-.3.3-.77 0-1.06L9.53 4.22a.75.75 0 0 0-1.06 0Z"
          fill="currentColor"
        />
      </svg>
    </span>
    Task #1
  </button>
</div>
`;

exports[`TimelineGroup > should render when timeline is collapsed 1`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <span
    className="fui-Text timeline-task ___rsnklj0_1dgriua fk6fouc fy9rknc fwrc4pm fdj6btp f17mccla f1w7gpdv f6juhto f1gl81tg f2jf649"
  >
    1
  </span>
  <div
    style={
      {
        "margin": "3px",
      }
    }
  >
    <div
      data-selected={false}
      data-testid="timeline-node"
    >
      TimelineNode - 
      rep-0
    </div>
  </div>
  <div
    style={
      {
        "margin": "3px",
      }
    }
  >
    <div
      data-selected={true}
      data-testid="timeline-node"
    >
      TimelineNode - 
      rep-1
    </div>
  </div>
  <div
    style={
      {
        "margin": "3px",
      }
    }
  >
    <div
      data-selected={false}
      data-testid="timeline-node"
    >
      TimelineNode - 
      rep-2
    </div>
  </div>
</div>
`;

exports[`TimelineGroup > should render with default props when timeline is expanded 1`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <button
    className="fui-Button r1alrhcs timeline-task ___rq2aul0_9etai30 fhovq9v f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb f1t94bn6 f1s2uweq fr80ssc f1ukrpxl fecsdlb fnwyq0v ft1hn21 fuxngvv fy5bs14 fsv2rcd f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1omzyqd f1dfjoow f1j98vj9 fj8yq94 f4xjyn1 f1et0tmh f9ddjv3 f1wi8ngl fh7ncta f1brlhvm f1sl3k7w fneth5b ft85np5 fy9rknc figsok6 fwrc4pm fazmxh"
    onClick={[Function]}
    style={
      {
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
    type="button"
  >
    <span
      className="fui-Button__icon rywnvv2 ___14u6ce1_ubi7uv0 f1nizpg2 fe5j1ua fjamq6b f64fuq3 fbaiahx"
    >
      <svg
        aria-hidden={true}
        className="fui-Icon-filled ___1vjqft9_1uxjcqz fjseox fez10in fg4l7m0"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.3 4.3a1 1 0 0 0 0 1.4l6.29 6.3-6.3 6.3a1 1 0 1 0 1.42 1.4l7-7a1 1 0 0 0 0-1.4l-7-7a1 1 0 0 0-1.42 0Z"
          fill="currentColor"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="fui-Icon-regular ___12fm75w_1vnroxd f1w7gpdv fez10in fg4l7m0"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.47 4.22c-.3.3-.3.77 0 1.06L15.19 12l-6.72 6.72a.75.75 0 1 0 1.06 1.06l7.25-7.25c.3-.3.3-.77 0-1.06L9.53 4.22a.75.75 0 0 0-1.06 0Z"
          fill="currentColor"
        />
      </svg>
    </span>
    Task #1
  </button>
</div>
`;

exports[`TimelineGroup > should render with different task number 1`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <button
    className="fui-Button r1alrhcs timeline-task ___rq2aul0_9etai30 fhovq9v f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb f1t94bn6 f1s2uweq fr80ssc f1ukrpxl fecsdlb fnwyq0v ft1hn21 fuxngvv fy5bs14 fsv2rcd f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1omzyqd f1dfjoow f1j98vj9 fj8yq94 f4xjyn1 f1et0tmh f9ddjv3 f1wi8ngl fh7ncta f1brlhvm f1sl3k7w fneth5b ft85np5 fy9rknc figsok6 fwrc4pm fazmxh"
    onClick={[Function]}
    style={
      {
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
    type="button"
  >
    <span
      className="fui-Button__icon rywnvv2 ___14u6ce1_ubi7uv0 f1nizpg2 fe5j1ua fjamq6b f64fuq3 fbaiahx"
    >
      <svg
        aria-hidden={true}
        className="fui-Icon-filled ___1vjqft9_1uxjcqz fjseox fez10in fg4l7m0"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.3 4.3a1 1 0 0 0 0 1.4l6.29 6.3-6.3 6.3a1 1 0 1 0 1.42 1.4l7-7a1 1 0 0 0 0-1.4l-7-7a1 1 0 0 0-1.42 0Z"
          fill="currentColor"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="fui-Icon-regular ___12fm75w_1vnroxd f1w7gpdv fez10in fg4l7m0"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.47 4.22c-.3.3-.3.77 0 1.06L15.19 12l-6.72 6.72a.75.75 0 1 0 1.06 1.06l7.25-7.25c.3-.3.3-.77 0-1.06L9.53 4.22a.75.75 0 0 0-1.06 0Z"
          fill="currentColor"
        />
      </svg>
    </span>
    Task #6
  </button>
</div>
`;

exports[`TimelineGroup > should render with empty repetitions 1`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <button
    className="fui-Button r1alrhcs timeline-task ___rq2aul0_9etai30 fhovq9v f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb f1t94bn6 f1s2uweq fr80ssc f1ukrpxl fecsdlb fnwyq0v ft1hn21 fuxngvv fy5bs14 fsv2rcd f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1omzyqd f1dfjoow f1j98vj9 fj8yq94 f4xjyn1 f1et0tmh f9ddjv3 f1wi8ngl fh7ncta f1brlhvm f1sl3k7w fneth5b ft85np5 fy9rknc figsok6 fwrc4pm fazmxh"
    onClick={[Function]}
    style={
      {
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
    type="button"
  >
    <span
      className="fui-Button__icon rywnvv2 ___14u6ce1_ubi7uv0 f1nizpg2 fe5j1ua fjamq6b f64fuq3 fbaiahx"
    >
      <svg
        aria-hidden={true}
        className="fui-Icon-filled ___1vjqft9_1uxjcqz fjseox fez10in fg4l7m0"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.3 8.3a1 1 0 0 1 1.4 0l6.3 6.29 6.3-6.3a1 1 0 1 1 1.4 1.42l-7 7a1 1 0 0 1-1.4 0l-7-7a1 1 0 0 1 0-1.42Z"
          fill="currentColor"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="fui-Icon-regular ___12fm75w_1vnroxd f1w7gpdv fez10in fg4l7m0"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.22 8.47c.3-.3.77-.3 1.06 0L12 15.19l6.72-6.72a.75.75 0 1 1 1.06 1.06l-7.25 7.25c-.3.3-.77.3-1.06 0L4.22 9.53a.75.75 0 0 1 0-1.06Z"
          fill="currentColor"
        />
      </svg>
    </span>
    Task #1
  </button>
</div>
`;

exports[`TimelineGroup > should render with no selected repetition 1`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <button
    className="fui-Button r1alrhcs timeline-task ___rq2aul0_9etai30 fhovq9v f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb f1t94bn6 f1s2uweq fr80ssc f1ukrpxl fecsdlb fnwyq0v ft1hn21 fuxngvv fy5bs14 fsv2rcd f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1omzyqd f1dfjoow f1j98vj9 fj8yq94 f4xjyn1 f1et0tmh f9ddjv3 f1wi8ngl fh7ncta f1brlhvm f1sl3k7w fneth5b ft85np5 fy9rknc figsok6 fwrc4pm fazmxh"
    onClick={[Function]}
    style={
      {
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
    type="button"
  >
    <span
      className="fui-Button__icon rywnvv2 ___14u6ce1_ubi7uv0 f1nizpg2 fe5j1ua fjamq6b f64fuq3 fbaiahx"
    >
      <svg
        aria-hidden={true}
        className="fui-Icon-filled ___1vjqft9_1uxjcqz fjseox fez10in fg4l7m0"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.3 4.3a1 1 0 0 0 0 1.4l6.29 6.3-6.3 6.3a1 1 0 1 0 1.42 1.4l7-7a1 1 0 0 0 0-1.4l-7-7a1 1 0 0 0-1.42 0Z"
          fill="currentColor"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="fui-Icon-regular ___12fm75w_1vnroxd f1w7gpdv fez10in fg4l7m0"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.47 4.22c-.3.3-.3.77 0 1.06L15.19 12l-6.72 6.72a.75.75 0 1 0 1.06 1.06l7.25-7.25c.3-.3.3-.77 0-1.06L9.53 4.22a.75.75 0 0 0-1.06 0Z"
          fill="currentColor"
        />
      </svg>
    </span>
    Task #1
  </button>
</div>
`;

exports[`TimelineGroup > should render with single repetition 1`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <button
    className="fui-Button r1alrhcs timeline-task ___rq2aul0_9etai30 fhovq9v f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb f1t94bn6 f1s2uweq fr80ssc f1ukrpxl fecsdlb fnwyq0v ft1hn21 fuxngvv fy5bs14 fsv2rcd f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1omzyqd f1dfjoow f1j98vj9 fj8yq94 f4xjyn1 f1et0tmh f9ddjv3 f1wi8ngl fh7ncta f1brlhvm f1sl3k7w fneth5b ft85np5 fy9rknc figsok6 fwrc4pm fazmxh"
    onClick={[Function]}
    style={
      {
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
    type="button"
  >
    <span
      className="fui-Button__icon rywnvv2 ___14u6ce1_ubi7uv0 f1nizpg2 fe5j1ua fjamq6b f64fuq3 fbaiahx"
    >
      <svg
        aria-hidden={true}
        className="fui-Icon-filled ___1vjqft9_1uxjcqz fjseox fez10in fg4l7m0"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.3 4.3a1 1 0 0 0 0 1.4l6.29 6.3-6.3 6.3a1 1 0 1 0 1.42 1.4l7-7a1 1 0 0 0 0-1.4l-7-7a1 1 0 0 0-1.42 0Z"
          fill="currentColor"
        />
      </svg>
      <svg
        aria-hidden={true}
        className="fui-Icon-regular ___12fm75w_1vnroxd f1w7gpdv fez10in fg4l7m0"
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.47 4.22c-.3.3-.3.77 0 1.06L15.19 12l-6.72 6.72a.75.75 0 1 0 1.06 1.06l7.25-7.25c.3-.3.3-.77 0-1.06L9.53 4.22a.75.75 0 0 0-1.06 0Z"
          fill="currentColor"
        />
      </svg>
    </span>
    Task #1
  </button>
</div>
`;
